src/pages/ItemManagePage.js

```diff
@@ -0,0 +1,288 @@
+    const { totalQuantity } = currentBatch;
+
+    const parsedSplitValue = parseInt(splitValue, 10);
+    let newObjectValues = {};
+
+    if (parsedSplitValue > totalQuantity) {
+      newObjectValues = { vvmStatus: false };
+      // Account for 0 & NaN (From entering a non-numeric character)
+    } else if (parsedSplitValue) {
+      const newBatchValues = { totalQuantity: parsedSplitValue, vvmStatus: false };
+      data.push(createRowObject(currentBatch, newBatchValues));
+      newObjectValues = { vvmStatus: true, totalQuantity: totalQuantity - parsedSplitValue };
+    }
+    this.updateObject(newObjectValues, { isModalOpen: false });
+  };
+
+  onApplyChanges = () => {
+    // TODO:
+    // Confirmation dialog
+    // Create inventory adjustments for any failed vvm status'
+    // Create repacks for any location changes
+    // Pop this page off the navigation stack.
+    // If more data is needed to create these repacks and inventory
+    // adjustments, they can easily be added to the row object with
+    // no side-effects. e.g. the ItemBatch itself or just
+    // some extra fields.
+  };
+
+  onDispose = () => {
+    // TODO:
+    // Open a modal to select the doses?
+    // Set the row to have some distinctive marker to show it
+    // is being disposed of.
+    // Set the dispose icon to one which will 'undelete' this row.
+  };
+
+  // Called on selecting a fridge in the fridge selection modal,
+  // just update the location of the currentBatch and close the modal.
+  onFridgeSelection = location => {
+    this.updateObject({ location }, { isModalOpen: false });
+  };
+
+  // Called on toggle the vvm toggle bar. If being set to PASS, will just update
+  // the object held in the closure to a PASS VVM status. Otherwise, opens a
+  // modal for a user to enter the doses affected and set the currentBatch.
+  onVvmToggle = currentBatch => ({ newState }) => {
+    if (!newState) return this.setState({ modalKey: 'vvmStatus', isModalOpen: true, currentBatch });
+    return this.setState({ currentBatch }, () => this.updateObject({ vvmStatus: true }));
+  };
+
+  // Method which controls all modals. Will open a modal corresponding to the
+  // modal key in renderModal and set the passed itemBatch as the current
+  // itemBatch. If called with no parameters, will clear the currentBatch
+  // and close the modal.
+  onModalUpdate = ({ modalKey, itemBatch } = {}) => () => {
+    if (modalKey) this.setState({ modalKey, isModalOpen: true, currentBatch: itemBatch });
+    else this.setState({ isModalOpen: false, currentBatch: null });
+  };
+
+  /**
+   * RENDER HELPERS
+   */
+  renderCell = (key, itemBatch) => {
+    const { location, vvmStatus } = itemBatch;
+    const modalUpdateProps = { modalKey: key, itemBatch };
+    const hasFridges = this.FRIDGES && this.FRIDGES.length > 0;
+    const usingFridge = vvmStatus !== false && hasFridges;
+    switch (key) {
+      default:
+        return itemBatch[key];
```

Comment:
can you use this component please for expiry date (set isEditable to false), it handles having null values

https://github.com/openmsupply/mobile/blob/9dbaf8a9f9115edc2583c3b23c715b81f34b4f8b/src/pages/SupplierInvoicePage.js#L211-L221

Reply:
dad042a1fa82e3be8006fcc4dbb2d64f273f7f05

Reply:
1b0cdefb68133463cad34740081999d16d504ffb